### PR TITLE
Adding name constants for LuaSyncedCtrl.cpp and LuaSyncedRead.cpp

### LINKED ISSUE #1530: Add named constants for magic -1/-2 for Lua table key/value stack indices
All over Lua code that iterates over tables there's these magic -1 and -2 constants used as stack indices. E.g.
https://github.com/beyond-all-reason/spring/blob/747f18b2222860ae264853aa062c5b3d9b164e25/rts/Lua/LuaSyncedRead.cpp#L2703-L2704
https://github.com/beyond-all-reason/spring/blob/747f18b2222860ae264853aa062c5b3d9b164e25/rts/Lua/LuaSyncedCtrl.cpp#L425-L426

Add (and use) some sort of named constants for them for self-documentation, e.g.
```cpp
static constexpr int LUA_TABLE_VALUE_INDEX = -1;
static constexpr int LUA_TABLE_KEY_INDEX   = -2;
```

### PR DESCRIPTION
**Pull request by 12345swordy**
Attempting to fix #1530 
CC @sprunk Let me know if I did something wrong in this PR

### CODE REVIEW

File: rts/Lua/LuaSyncedCtrl.cpp [hunk 1]
```diff
@@ -5787,7 +5787,7 @@ int LuaSyncedCtrl::SetHeightMapFunc(lua_State* L)
 
 	if (error != 0) {
 		LOG_L(L_ERROR, "Spring.SetHeightMapFunc: error(%i) = %s",
-				error, lua_tostring(L, -1));
+				error, lua_tostring(L, LUA_TABLE_VALUE_INDEX));
```
Comment (sprunk) [inline rts/Lua/LuaSyncedCtrl.cpp:5790]: This one isn't a table index, it's not within a loop iterating over a table.

File: rts/Lua/LuaSyncedCtrl.cpp [hunk 2]
```diff
@@ -6950,8 +6950,8 @@ static bool ParseNamedInt(lua_State* L, const string& key,
 	if (key != name) {
 		return false;
 	}
-	if (lua_isnumber(L, -1)) {
-		value = lua_toint(L, -1);
+	if (lua_isnumber(L, LUA_TABLE_VALUE_INDEX)) {
```
Comment (sprunk) [inline rts/Lua/LuaSyncedCtrl.cpp:6953]: ditto

**Comment by sprunk** [comment]
Generally anything not within a loop that starts like this:
```cpp
for (lua_pushnil(L); lua_next(L, SOME_NUMBER) != 0; lua_pop(L, 1)) {
```
is not applicable.



A caveat is that sometimes it's a var passed from outside the loop. For example
https://github.com/beyond-all-reason/spring/blob/ce86af7df7524a799d804e903e7851a6f7c85ae1/rts/Lua/LuaUnsyncedCtrl.cpp#L1296
`unitIndexInTable` is passed -1 from here:
https://github.com/beyond-all-reason/spring/blob/ce86af7df7524a799d804e903e7851a6f7c85ae1/rts/Lua/LuaUnsyncedCtrl.cpp#L1320

File: rts/Lua/LuaSyncedCtrl.cpp [hunk 3]
```diff
@@ -1465,9 +1465,9 @@ static inline void ParseCobArgs(
 			lua_rawgeti(L, a, 1);
 			lua_rawgeti(L, a, 2);
 
-			if (lua_isnumber(L, -2) && lua_isnumber(L, -1)) {
-				const int x = lua_toint(L, -2);
-				const int z = lua_toint(L, -1);
+			if (lua_isnumber(L, LUA_TABLE_KEY_INDEX) && lua_isnumber(L, LUA_TABLE_VALUE_INDEX)) {
+				const int x = lua_toint(L, LUA_TABLE_KEY_INDEX);
+				const int z = lua_toint(L, LUA_TABLE_VALUE_INDEX);
```
Comment (sprunk) [inline rts/Lua/LuaSyncedCtrl.cpp:1470]: not key/value

File: rts/Lua/LuaSyncedRead.cpp [hunk 4]
```diff
@@ -1597,10 +1597,10 @@ int LuaSyncedRead::GetMapStartPositions(lua_State* L)
...
 
 		lua_createtable(L, 3, 0);
-		lua_pushnumber(L, pos.x); lua_rawseti(L, -2, 1);
-		lua_pushnumber(L, pos.y); lua_rawseti(L, -2, 2);
-		lua_pushnumber(L, pos.z); lua_rawseti(L, -2, 3);
-		lua_rawseti(L, -2, 1 + teamNum); // [i] = {x,y,z}
+		lua_pushnumber(L, pos.x); lua_rawseti(L, LUA_TABLE_KEY_INDEX, 1);
+		lua_pushnumber(L, pos.y); lua_rawseti(L, LUA_TABLE_KEY_INDEX, 2);
+		lua_pushnumber(L, pos.z); lua_rawseti(L, LUA_TABLE_KEY_INDEX, 3);
+		lua_rawseti(L, LUA_TABLE_KEY_INDEX, 1 + teamNum); // [i] = {x,y,z}
```
Comment (sprunk) [inline rts/Lua/LuaSyncedRead.cpp:1603]: not key, just a normal -2

File: rts/Lua/LuaSyncedCtrl.cpp [hunk 5]
```diff
@@ -7022,10 +7022,10 @@ static bool ParseCommandDescription(lua_State* L, int table,
 	}
 
 	for (lua_pushnil(L); lua_next(L, table) != 0; lua_pop(L, 1)) {
-		if (!lua_israwstring(L, -2))
+		if (!lua_israwstring(L, LUA_TABLE_KEY_INDEX))
 			continue;
 
-		const string key = lua_tostring(L, -2);
+		const string key = lua_tostring(L, -1);
```
[the remark this answers is not in the document]
Comment (12345swordy) [reply rts/Lua/LuaSyncedCtrl.cpp]: Woops! I revert it incorrectly, its fixed now.

**Comment by 12345swordy** [comment]
What exactly is the hold up here?

**Comment by lhog** [comment]
I think it can be merged after another thorough review searching for possible replacement errors

**Comment by 12345swordy** [comment]
@sprunk done

File: rts/Lua/LuaSyncedCtrl.cpp [hunk 6]
```diff
@@ -6038,7 +6038,7 @@ int LuaSyncedCtrl::SetOriginalHeightMapFunc(lua_State* L)
 
 	if (error != 0) {
 		LOG_L(L_ERROR, "Spring.SetOriginalHeightMapFunc: error(%i) = %s",
-				error, lua_tostring(L, -1));
+				error, lua_tostring(L, LUA_TABLE_VALUE_INDEX));
```
Comment (sprunk) [inline rts/Lua/LuaSyncedCtrl.cpp:6041]: Not table value (it's the error object returned by `lua_pcall` in line 6036)

File: rts/Lua/LuaSyncedRead.cpp
Comment (sprunk) [inline rts/Lua/LuaSyncedRead.cpp:1]: -2/-1 inside `lua_rawseti` never refers to table key/value. At a glance this means all changes in this file (with one exception noted below) are to be dropped.

File: rts/Lua/LuaSyncedRead.cpp [hunk 7]
```diff
@@ -2701,7 +2701,7 @@ int LuaSyncedRead::GetTeamUnitsByDefs(lua_State* L)
 		const int tableIdx = 2;
 
 		for (lua_pushnil(L); lua_next(L, tableIdx) != 0; lua_pop(L, 1)) {
-			if (!lua_isnumber(L, -1))
+			if (!lua_isnumber(L, LUA_TABLE_VALUE_INDEX))
```
Comment (sprunk) [inline rts/Lua/LuaSyncedRead.cpp:2704]: Here is the exception.

**Comment by 12345swordy** [comment]
@sprunk done

**Review by sprunk — APPROVED** [review]
I haven't found further errors, i.e. there aren't false positives. There are some false negatives (i.e. left as magic -2/-1) but that is fine, can be added later.

**Pull request merged**